Daily cron job fails because of a KeyError Exception

See [here](https://github.com/MobilityData/mobility-database-catalogs/runs/7511188083?check_suite_focus=true#step:5:85) 

```
Traceback (most recent call last):
  File "/home/runner/work/_temp/9dc6b242-1171-48be-9141-1321feb026ab.py", line 22, in <module>
    api_source_secrets = json.loads(os.environ[API_SOURCE_SECRETS])
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/os.py", line 6[79](https://github.com/MobilityData/mobility-database-catalogs/runs/7511188083?check_suite_focus=true#step:5:80), in __getitem__
    raise KeyError(key) from None
KeyError: 'API_SOURCE_SECRETS'
Error: Process completed with exit code 1.
```

--- a/.github/workflows/direct_download_urls_test_for_sources.yml
+++ b/.github/workflows/direct_download_urls_test_for_sources.yml
@@ -126,6 +126,8 @@ jobs:
           if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
       - name: Validate and download the datasets
         shell: python
+        env:
+          API_SOURCE_SECRETS: ${{ secrets.API_SOURCE_SECRETS }}
         run: |
           import os
           import json
@@ -155,7 +157,7 @@ jobs:
               url = job_json[DIRECT_DOWNLOAD]
               authentication_type = job_json[AUTHENTICATION_TYPE]
               api_key_parameter_name = job_json[API_KEY_PARAMETER_NAME]
-              api_key_parameter_value = api_source_secrets[base]
+              api_key_parameter_value = api_source_secrets.get(base)
 
               # Download the dataset
               zip_path = os.path.join(ROOT, DATASETS, f"{base}.zip")

--- a/.github/workflows/store_latest_datasets_cronjob.yml
+++ b/.github/workflows/store_latest_datasets_cronjob.yml
@@ -3,6 +3,8 @@ name: Store latest datasets cronjob
 on:
   schedule:
     - cron: "0 0 * * *"
+  pull_request:
+    branches: [ main ]
 
 jobs:
   get-urls:

--- a/.github/workflows/store_latest_datasets_cronjob.yml
+++ b/.github/workflows/store_latest_datasets_cronjob.yml
@@ -141,6 +141,8 @@ jobs:
           pip install wheel requests
       - name: Validate and download the datasets
         shell: python
+        env:
+          API_SOURCE_SECRETS: ${{ secrets.API_SOURCE_SECRETS }}
         run: |
           import os
           import json
@@ -172,7 +174,7 @@ jobs:
               url = job_json[DIRECT_DOWNLOAD]
               authentication_type = job_json[AUTHENTICATION_TYPE]
               api_key_parameter_name = job_json[API_KEY_PARAMETER_NAME]
-              api_key_parameter_value = api_source_secrets[base]
+              api_key_parameter_value = api_source_secrets.get(base)
 
               # Download the dataset
               zip_path = os.path.join(ROOT, DATASETS, FOLDER, f"{base}.zip")

--- a/.github/workflows/store_latest_datasets_cronjob.yml
+++ b/.github/workflows/store_latest_datasets_cronjob.yml
@@ -3,8 +3,6 @@ name: Store latest datasets cronjob
 on:
   schedule:
     - cron: "0 0 * * *"
-  pull_request:
-    branches: [ main ]
 
 jobs:
   get-urls:

